#51 Add watchlist

lfranke42 opened the pull request:
fixes #51 

<img width="1192" height="1682" alt="localhost_3000_watchlist" src="https://github.com/user-attachments/assets/14e8f08f-da46-4fe0-8719-1db63bde00df" />
<img width="1921" height="1018" alt="Screenshot 2025-11-29 at 19 03 42" src="https://github.com/user-attachments/assets/3c1632d0-c658-4db3-abac-f9a605094062" />
<img width="1921" height="1018" alt="Screenshot 2025-11-29 at 19 09 18" src="https://github.com/user-attachments/assets/fedc582c-7c64-4592-bdec-9f74d8c04aaa" />

bruderj15 reviewed (commented):
Looks great! How difficult would classical red hearts and maybe even some subtle animation be? (think instagram)

> I also added NavLinks to the header, but this probably needs some more attention.

First impression is to group by public/personal here.
Towards the left of the search-bar I'd do all public things (shop, ...).
Towards the right (where the User Profile-Icon is) I'd put all personal things (watchlist, search-filters, ...).

bruderj15 reviewed (approved):
Simply lovely!

lfranke42 commented on src/components/watchlist/__tests__/WatchlistResults.test.tsx:
```diff
@@ -0,0 +1,307 @@
+            setInfiniteQueryMock({ items: [] });
+            renderWithQueryClient(<WatchlistResults />);
+
+            expect(screen.getByText("Keine Gegenstände gefunden")).toBeInTheDocument();
```
Do we have a consensus on the "Artikel" wording? Then I'd change everything over

lfranke42 commented:
I adjusted the wording, moved the navbar entry to the right and included a mobile layout option. Images at the the top have been updated. Should be properly ready for review now.

BauZee commented on src/components/watchlist/WatchlistResults.tsx:
What do you think about making the image full width in the mobile view, just like the ItemDetailInfo? On the one hand, it would be more consistent, and on the other hand, I think it would look a little better.
<img width="830" height="886" alt="image" src="https://github.com/user-attachments/assets/1cb44885-22e1-4109-a359-395b3fadff0f" />

lfranke42 replied:
<img width="1428" height="2368" alt="Image" src="https://github.com/user-attachments/assets/7c9ef7e5-f90f-4caf-9255-f9ce51b075f7" />

BauZee commented on src/i18n/locales/de/translation.json:
```diff
@@ -198,5 +200,17 @@
+    "watchlist": {
+        "title": "Meine Merkliste",
+        "totalElements": "{{count}} Artikel",
+        "loadingError": "Die Merkliste kann zur Zeit nicht erreicht werden. Bitte versuchen Sie es später erneut.",
```
```suggestion
        "loadingError": "Die Merkliste kann zurzeit nicht erreicht werden. Bitte versuchen Sie es später erneut.",
```

lfranke42 replied:
fixed

lfranke42 commented on src/components/watchlist/WatchlistResults.tsx:
```diff
@@ -0,0 +1,147 @@
+export function WatchlistResults() {
+    const { ref, inView } = useInView();
+    const { t } = useTranslation();
+    const { data, isPending, error, fetchNextPage, hasNextPage, isFetchingNextPage } =
```
fixed

lfranke42 commented on src/components/watchlist/WatchlistResults.tsx:
```diff
@@ -0,0 +1,147 @@
+                    },
+                });
+
+                if (result.error) {
```
fixed

lfranke42 commented on src/i18n/locales/de/translation.json:
```diff
@@ -198,5 +200,17 @@
+        "totalElements": "{{count}} Artikel",
+        "loadingError": "Die Merkliste kann zur Zeit nicht erreicht werden. Bitte versuchen Sie es später erneut.",
+        "loadingMore": "Lade neue Ergebnisse...",
+        "allLoaded_one": "Sie haben {{count}} Artikel ihrer Merkliste gesehen.",
```
fixed

lfranke42 commented on src/i18n/locales/de/translation.json:
```diff
@@ -198,5 +200,17 @@
+        "loadingError": "Die Merkliste kann zur Zeit nicht erreicht werden. Bitte versuchen Sie es später erneut.",
+        "loadingMore": "Lade neue Ergebnisse...",
+        "allLoaded_one": "Sie haben {{count}} Artikel ihrer Merkliste gesehen.",
+        "allLoaded_other": "Sie haben alle {{count}} Artikel ihrer Merkliste gesehen.",
```
fixed

lfranke42 commented on src/client/types.gen.ts:
```diff
@@ -1,7 +1,7 @@
 
 export type ClientOptions = {
-    baseUrl: string;
+    baseUrl: 'https://api.aura-historia.com' | (string & {});
```
yes it's correct

BauZee commented on src/components/watchlist/WatchlistResults.tsx:
Then create tickets, either grouped together or individually, for the issues that you are not currently working on here, so that we or copilot can take care of them.
